cookbook/knowledge/vector_db/README.md

```diff
@@ -0,0 +1,309 @@
+- Good SQL ecosystem integration
+- ACID transactions
+- Cost-effective for small to medium scale
+
+```python
+from agno.knowledge.knowledge import Knowledge
+from agno.vectordb.pgvector import PgVector, SearchType
+
+# Basic PgVector setup
+db_url = "postgresql+psycopg://ai:ai@localhost:5532/ai"
+
+knowledge = Knowledge(
+    vector_db=PgVector(
+        table_name="company_knowledge",
+        db_url=db_url,
+        search_type=SearchType.hybrid,  # Vector + keyword search
+        dimensions=1536,  # OpenAI embedding dimensions
+    )
+)
+
+# Add content with metadata
+knowledge.add_content(
+    path="company_docs/",
+    metadata={
+        "department": "engineering",
+        "document_type": "technical_spec",
+        "access_level": "internal"
+    }
+)
+
+# Advanced PgVector configuration
+advanced_vector_db = PgVector(
+    table_name="advanced_knowledge",
+    db_url=db_url,
+    search_type=SearchType.hybrid,
+    # Performance optimizations
+    connection_pool_size=20,
+    max_overflow=10,
+    index_type="ivfflat",  # or "hnsw" for larger datasets
+    lists=100,
+    # Metadata indexing
+    metadata_indexes=["department", "document_type", "created_at"]
+)
+```
+
+**Best practices**:
+- Use connection pooling for production
+- Create indexes on frequently filtered metadata fields
+- Choose appropriate index type (ivfflat vs hnsw) based on dataset size
+- Monitor query performance and optimize accordingly
+
+**Examples**:
+- **[Basic PgVector](./pgvector/pgvector_db.py)** - Simple setup and usage
+- **[Async PgVector](./pgvector/async_pg_vector.py)** - Asynchronous operations
+- **[Hybrid Search](./pgvector/pgvector_hybrid_search.py)** - Combined vector/keyword search
+
+### 2. Cloud Vector Databases
+
+#### Pinecone (`pinecone_db/`)
+
+**Strengths**:
+- Fully managed service
+- Automatic scaling
+- High availability
+
+```python
+from agno.vectordb.pinecone import PineconeDB
+
+pinecone_db = PineconeDB(
+    name="company-knowledge",
+    dimension=1536,
+    metric="cosine",
+    environment="us-west1-gcp-free",  # Choose your region
+    api_key="your_pinecone_api_key"
+)
+
+knowledge = Knowledge(
+    vector_db=pinecone_db,
+    name="Pinecone Knowledge Base"
+)
+```
+
+#### Qdrant (`qdrant_db/`)
+
+**Strengths**:
+- Good filtering and payload support
+- High performance
+- Cloud or self-hosted options
+
+```python
+from agno.vectordb.qdrant import Qdrant
+
+# Qdrant Cloud
+qdrant_cloud = Qdrant(
+    collection="company_knowledge",
+    url="https://your-cluster.qdrant.io:6333",
+    api_key="your_qdrant_api_key",
+    distance="Cosine"
+)
+
+# Self-hosted Qdrant
+qdrant_local = Qdrant(
+    collection="local_knowledge",
+    url="http://localhost:6333",
+    distance="Cosine",
+    # Advanced configuration
+    vector_size=1536,
+    on_disk_payload=True,  # Store large payloads on disk
+    replication_factor=2,   # For high availability
+)
+```
+
+### 3. Enterprise Solutions
```

Comment:
Why is this enterprise? Lets rather not group them and just have a list of DBs. We also don't really need examples in the READMEs. Can just link to the cookbooks for the specific DBs